### PR TITLE
Add a new transaction-messages package

### PR DESCRIPTION
**Pull request by mcintyre94**
This package will contain the code currently in @solana/transactions that involved unsigned transaction messages

For now it has the types TransactionMessage and BaseTransactionMessage (copied from Transaction/BaseTransaction), and the createTransactionMessage function (copied from createTransaction)

It will also have the code to manipulate a transaction message, eg adding a fee payer/lifetime etc, and the code to compile a message.

It will become a dependency of `@solana/transactions`

### CODE REVIEW

**Comment by mcintyre94** [comment]
* **#2423** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2423?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @mcintyre94 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/transaction-messages/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+    "$schema": "https://json.schemastore.org/tsconfig",
+    "compilerOptions": {
+        "lib": ["DOM", "ES2017", "ES2022.Error"]
+    },
+    "display": "@solana/transactions",
+    "extends": "@solana/tsconfig/base.json",
```
Comment (steveluscher) [inline packages/transaction-messages/tsconfig.json]: I screwed you in #2355.
```suggestion
    "extends": "../tsconfig/base.json",
```

File: packages/transaction-messages/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+    "$schema": "https://json.schemastore.org/tsconfig",
+    "compilerOptions": {
+        "lib": ["DOM", "ES2017", "ES2022.Error"]
+    },
+    "display": "@solana/transactions",
```
Comment (steveluscher) [inline packages/transaction-messages/tsconfig.json]:
```suggestion
    "display": "@solana/transaction-messages",
```

File: packages/transaction-messages/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+    "$schema": "https://json.schemastore.org/tsconfig",
+    "compilerOptions": {
+        "lib": ["DOM", "ES2017", "ES2022.Error"]
```
Comment (steveluscher) [inline packages/transaction-messages/tsconfig.json:4]: Check if you need all of these.

**Review by steveluscher — APPROVED** [review]
If it's still possible to do a little `git-cp` action so that we keep the history, maybe give it a try? I think you can use this to make a mess of commits in this stack, and then let Graphite squash them.

Maybe.

https://gitlab.inria.fr/-/snippets/520

**Review by lorisleiva — APPROVED** [review]

**Comment by mcintyre94** [comment]
> If it's still possible to do a little git-cp action so that we keep the history, maybe give it a try? I think you can use this to make a mess of commits in this stack, and then let Graphite squash them.

Testing this locally, it does work but as soon as you squash the commits it loses the history. Or more likely I'm selecting the wrong resolutions on rebase errors when I squash them to keep the history. But given those rebase errors on squash, it makes me think it's probably not going to work with a github/graphite squash + merge unfortunately.

**Comment by mcintyre94** [comment]
Okay cool, I think I've found the way to do it: rebase, edit the commit before the file is added, run the copy script there (this adds a bunch of new commits between the edited commit and the next one), rebase continue, fix the merge conflict by accepting incoming (this applies the changes between the copy and the new version)

I've not been able to rebase to squash these commits locally because they have conflicts and I don't know how to resolve them properly to maintain the history. So I'll just leave lots of commits in the PR and make the message sensible when I squash. But Github has the right history on this PR, and I'll assume/hope that it maintains it when I squash + merge. If it doesn't then I'll ditch this plan for the other PRs in the stack.

File: packages/transaction-messages/tsconfig.json (hunk 3 above)
Comment (mcintyre94) [reply packages/transaction-messages/tsconfig.json]: Removed them all for now, I'll add whatever's needed as I go

**Comment by mcintyre94** [comment]
Sad times, the github squash + merge kills the history :(

History is on this PR: https://github.com/solana-labs/solana-web3.js/blame/3bb1f9289d6354e76eff6c500b603d2848f811b9/packages/transaction-messages/src/create-transaction-message.ts

But after merging it's lost it: https://github.com/solana-labs/solana-web3.js/commits/master/packages/transaction-messages/src/create-transaction-message.ts

**Pull request merged**